lib/spack/spack/environment/environment.py

```diff
@@ -1837,6 +1837,7 @@ def _install_log_links(self, spec):
                 )
                 if os.path.lexists(build_log_link):
                     os.remove(build_log_link)
+                fs.touchp(spec.package.build_log_path)
```

Comment:
(question) This function runs after all packages are installed, so I'm surprised this is needed. Do things seem to fail universally without this? Or is it just in special cases?

Reply:
I think the failures were in the `uninstall.py` unit test, so maybe there's no packages installed in that case? Maybe the touch should either be in the unit test itself or protected by an `if not os.path.exists(spec.package.build_log_path):`.

Reply:
I think doing this in `conftest._pkg_install_fn` would be more clear.

Reply:
@scheibelp I backed this commit out and changed `conftest._pkg_install_fn` to 

```
 def _pkg_install_fn(pkg, spec, prefix):
     # sanity_check_prefix requires something in the install directory
     mkdirp(prefix.bin)
+    if not os.path.exists(spec.package.build_log_path):  
+        touchp(spec.package.build_log_path) 
```

Is that what you meant? It doesn't fix the unit test failures. Any suggestions?